Reproducible test failure in TestFloatVectorSimilarityQuery.testBoostQuery

### Description

```
org.apache.lucene.search.TestFloatVectorSimilarityQuery > testBoostQuery FAILED
    java.lang.AssertionError: expected:<29> but was:<8>
        at __randomizedtesting.SeedInfo.seed([3F71A3D0594CEBC0:A66FB115DC7DBC27]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.apache.lucene.search.BaseVectorSimilarityQueryTestCase.testBoostQuery(BaseVectorSimilarityQueryTestCase.java:362)
        at org.apache.lucene.search.TestFloatVectorSimilarityQuery.testBoostQuery(TestFloatVectorSimilarityQuery.java:26)
```


```
gradlew test --tests TestFloatVectorSimilarityQuery.testBoostQuery -Dtests.seed=3F71A3D0594CEBC0 -Dtests.nightly=true -Dtests.locale=de-Latn-DE -Dtests.timezone=Pacific/Wallis -Dtests.asserts=true -Dtests.file.encoding=UTF-8
```

### Gradle command to reproduce

_No response_

--- a/lucene/core/src/test/org/apache/lucene/search/BaseVectorSimilarityQueryTestCase.java
+++ b/lucene/core/src/test/org/apache/lucene/search/BaseVectorSimilarityQueryTestCase.java
@@ -17,7 +17,14 @@
 package org.apache.lucene.search;
 
 import java.io.IOException;
-import java.util.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.IntStream;
 import org.apache.lucene.document.Document;
 import org.apache.lucene.document.Field;

--- a/lucene/core/src/test/org/apache/lucene/search/BaseVectorSimilarityQueryTestCase.java
+++ b/lucene/core/src/test/org/apache/lucene/search/BaseVectorSimilarityQueryTestCase.java
@@ -17,13 +17,7 @@
 package org.apache.lucene.search;
 
 import java.io.IOException;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Set;
+import java.util.*;
 import java.util.stream.IntStream;
 import org.apache.lucene.document.Document;
 import org.apache.lucene.document.Field;
@@ -356,11 +350,18 @@ abstract class BaseVectorSimilarityQueryTestCase<
       Query query2 = new BoostQuery(query1, boost);
       ScoreDoc[] scoreDocs2 = searcher.search(query2, numDocs).scoreDocs;
 
-      // Check that all docs are identical, with boosted scores
+      // Check that original scores and boosted scores are equal considering the delta to account
+      // for floating point precision limitations. Don't take the exact result order into
+      // consideration as for small original scores with tiny differences
+      // the boosted scores can be become the same, which might affect the result order.
       assertEquals(scoreDocs1.length, scoreDocs2.length);
       for (int i = 0; i < scoreDocs1.length; i++) {
-        assertEquals(scoreDocs1[i].doc, scoreDocs2[i].doc);
-        assertEquals(boost * scoreDocs1[i].score, scoreDocs2[i].score, delta);
+        int idx = i;
+        Optional<ScoreDoc> boostedDoc =
+            Arrays.stream(scoreDocs2).filter(d -> d.doc == scoreDocs1[idx].doc).findFirst();
+
+        assertTrue(boostedDoc.isPresent());
+        assertEquals(boost * scoreDocs1[i].score, boostedDoc.get().score, delta);
       }
     }
   }

